### PR TITLE
New package: yazi-0.2.5

### LINKED ISSUE #46843: Package request: yazi
### Package name

yazi

### Package homepage

https://github.com/sxyazi/yazi

### Description

Yazi  is a terminal file manager written in Rust, based on non-blocking async I/O. It aims to provide an efficient, user-friendly, and customizable file management experience.

### Does the requested package meet the package requirements?

Compiled

### Is the requested package released?

Yes

### PR DESCRIPTION
**Pull request by Bnyro**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - armv6l-musl

closes #46843

### CODE REVIEW

**Comment by rookiejet** [comment]
Can you add zsh completion as well? It's in the `_yazi` file along with the other completions.  Thanks for the package!

**Comment by Bnyro** [comment]
> Can you add zsh completion as well? It's in the `_yazi` file along with the other completions. Thanks for the package!
>
Which folder do you mean? The `yazi` folder only contains `_yazi.ps1
yazi.bash
yazi.elv
yazi.fish
yazi.nu
yazi.ts` but no zsh completions.

**Comment by rookiejet** [comment]
It is [generated along with those](https://github.com/sxyazi/yazi/blob/396f60d9e026f6e5e83d8717a47e6d9c98fc8068/yazi-config/build.rs#L21).

Adding `vcompletion config/completions/_yazi zsh` works from my local testing.

**Comment by Bnyro** [comment]
Okay thanks :+1:

I've just been wondering because the completion file doesn't have a `.zsh`extension.

**Comment by realcharmer** [comment]
I've built this on x86-glibc and all works well. Thanks for the package.

**Comment by Anidetrix** [comment]
yazi uses vendored luajit52 as seen [here](https://github.com/sxyazi/yazi/blob/v0.1.5/plugin/Cargo.toml#L14)

from [mlua README](https://github.com/khvzak/mlua#compiling)
> mlua also supports vendored lua/luajit using the auxiliary crates [lua-src](https://crates.io/crates/lua-src) and [luajit-src](https://crates.io/crates/luajit-src). Just enable the vendored feature and cargo will automatically build and link specified lua/luajit version. This is the easiest way to get started with mlua.

which means yazi will bundle it's own luajit52, so changes to plugin/Cargo.toml and lua52-devel dependency should be removed I believe

**Comment by Bnyro** [comment]
And I don't think there's a reason why yazi should bundle its own lua version instead of using the system one (which saves storage space in general)...

**Comment by Anidetrix** [comment]
> Cross compilation won't work then, try it yourself if you don't believe me.

Ah, got it. Will test it to see for myself to see if it can be fixed.

> And I don't think there's a reason why yazi should bundle its own lua version instead of using the system one (which saves storage space in general)...

As you said yourself: we can't use luajit52 because it's not in the repos. So if we don't have it in the repo (and also being difficult to support in the repo due to LuaJIT's rolling release model)  - don't see the reason to not allow it. But if it breaks the cross-compilation then the slower lua is definitely better than a broken build. If so, you do need to remove the `vendored` feature from the `plugin/Cargo.toml` file as well, cause right now it still bundles the normal lua52 anyway.

In any case, thank you, would love to see this package in the repo.

**Comment by sxyazi** [comment]
Hi, I have released Yazi v0.2.1, removed luajit and replaced it with lua5.4 to address build issues (and exception handling problems) on multiple platforms. Hope this will be helpful, thank you for bringing Yazi to Void Linux! ❤️️

**Comment by Bnyro** [comment]
> Hi, I have released Yazi v0.2.1, removed luajit and replaced it with lua5.4 to address build issues (and exception handling problems) on multiple platforms. Hope this will be helpful, thank you for bringing Yazi to Void Linux! ❤️️
>
Thank you for the update! Everything seems to build fine now with 0.2.1, however it seems like the auto completions for bash/zsh/fish are no longer there?

**Comment by Anidetrix** [comment]
> it seems like the auto completions for bash/zsh/fish are no longer there

@Bnyro I fixed it in my local template by adding:
```sh
export YAZI_GEN_COMPLETIONS=true
```
into the main body of the template

also adding:
```sh
export VERGEN_GIT_SHA="${version}_${revision}"
```
fixes `yazi -V` output

**Comment by sxyazi** [comment]
> Thank you for the update! Everything seems to build fine now with 0.2.1

Awesome!

> however it seems like the auto completions for bash/zsh/fish are no longer there?

`export YAZI_GEN_COMPLETIONS=1` before building you'll get it. 😉

File: srcpkgs/yazi/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+homepage="https://yazi-rs.github.io"
+distfiles="https://github.com/sxyazi/yazi/archive/refs/tags/v${version}.tar.gz"
+checksum=c821979b1ae08e7bcf2936635bfcd25715e74f240a6c4a2dbeb4ba9ce0d8be8b
+
+export VERGEN_GIT_SHA="${version}_${revision}"
+# enable the generation of shell auto completions
+export YAZI_GEN_COMPLETIONS=true
+
+post_install() {
+	vcompletion config/completions/yazi.bash bash
```
Comment (Anidetrix) [inline srcpkgs/yazi/template:22]: new path is `yazi-config` instead of `config`

File: srcpkgs/yazi/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'yazi'
+pkgname=yazi
+version=0.2.1
+revision=1
+build_style=cargo
+make_install_args="--path yazi-fm"
+hostmakedepends="pkg-config"
+makedepends="oniguruma-devel lua54-devel"
+depends="nerd-fonts-symbols-ttf"
+short_desc="Blazing fast terminal file manager"
```
Comment (sxyazi) [inline srcpkgs/yazi/template:10]: How about using this full description "Blazing fast terminal file manager written in Rust, based on async I/O"?

**Comment by LinArcX** [comment]
Why is this package still in progress? something is missing?

**Comment by melroy89** [comment]
Hopefully it can be merged?

**Comment by sxyazi** [comment]
Maybe we can merge this PR (v0.2.1) first?

**Comment by zen0bit** [comment]
0.2.4 is out...

**Review by zen0bit — APPROVED** [review]

**Review by sxyazi — APPROVED** [review]

File: srcpkgs/yazi/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'yazi'
+pkgname=yazi
+version=0.2.4
+revision=1
+build_style=cargo
+make_install_args="--path yazi-fm"
+hostmakedepends="pkg-config"
+makedepends="oniguruma-devel lua54-devel"
+depends="nerd-fonts-symbols-ttf"
+short_desc="Blazing fast terminal file manager written in Rust, based on async I/O"
```
Comment (classabbyamp) [inline srcpkgs/yazi/template:10]:
```suggestion
short_desc="Blazing fast terminal file manager, based on async I/O"
```

**Comment by toomyem** [comment]
Will it be merged eventually? :smile:

**Comment by manfred3000** [comment]
In case that `yazi`'s disclaimer 

> Note: Yazi is currently in heavy development and may be unstable. The API is subject to change without prior notice.

might (still) have a deterrent effect on adding yazi to Void's repo, I for one have been using it for a few months (running Void for a month or so, now) and it hasn't randomly deleted any files, yet... :P

It's true that it does require some attention when it comes to manually adapting config changes, but I hope that `yazi` will be part of the Void soon!

Anyway, I appreciate Void's stability and trust the devs' decision making...

Thanks @sxyazi for your work!

**Comment by sxyazi** [comment]
Yazi will **NOT** randomly delete user files. 

On the contrary, we've made extensive security efforts. For example:

- The default behavior for deletion ([`remove` command](https://yazi-rs.github.io/docs/configuration/keymap/#manager.remove)) is to move files to the trash instead of permanently deleting them. If users want to permanently delete files, they need to manually specify `--permanently`, and there's a confirmation prompt for every deletion even trashing.
- For shell commands like [`shell "nvim"`](https://yazi-rs.github.io/docs/configuration/keymap/#manager.shell), by default, a dialog box pops up displaying the shell command template passed by the user for confirmation, and the command isn't executed. It's only executed if the user explicitly specifies `shell "nvim" --confirm`. 
- Splitting the commands/keys for [opening](https://yazi-rs.github.io/docs/configuration/keymap/#manager.open) and [entering directories](https://yazi-rs.github.io/docs/configuration/keymap/#manager.enter) also includes security considerations. Users navigate between files using Vim-like `hjkl` keys, and when they're sure they want to open/run the selected file, they use `<Enter>` instead of using `l` like in other file managers, which is used for both directory navigation and opening files. 
- We've implemented a [nested selection conflict detection](https://github.com/sxyazi/yazi/issues/688) to prevent users from selecting `/a/b` while also selecting `/a`, as their mutual inclusion could lead to file confusion.
- In the previous version, we also added [detection for the same directory](https://github.com/sxyazi/yazi/issues/894) to prevent users from copying `./foo` as `./foo/foo`, `./foo/foo/foo`, which would result in recursive copying. 

Having said all that, I just want to emphasize that **security is always Yazi's top priority**, and this shouldn't be a reason to prevent this PR from being merged (if it is).

**Comment by toomyem** [comment]
What is holding back this PR from being merged?

**Pull request merged**